axum-extra/src/protobuf.rs

```diff
@@ -0,0 +1,303 @@
+//! Protocol Buffer extractor and response.
+
+use axum::{
+    async_trait,
+    body::{Bytes, HttpBody},
+    extract::{rejection::BytesRejection, FromRequest, RequestParts},
+    response::{IntoResponse, Response},
+    BoxError,
+};
+use http::StatusCode;
+use prost::Message;
+use std::ops::{Deref, DerefMut};
+
+/// Protocol Buffer extractor and response.
+///
+/// When used as an extractor, it can decode request bodies into some type that
+/// implements [`prost::Message`]. The request will be rejected (and a [`ProtoBufRejection`] will
+/// be returned) if:
+///
+/// - The body couldn't be decoded into the target Protocol Buffer.
+/// - Buffering the request body fails.
+///
+/// See [`ProtoBufRejection`] for more details.
+///
+/// # Extractor example
```

Comment:
For consistency with [`JsonLines`](https://docs.rs/axum-extra/latest/axum_extra/json_lines/struct.JsonLines.html)

```suggestion
/// # As extractor
```

Reply:
Probably also worth documenting that the extractor doesn't care about the `content-type`.